landoapi/models/landing_job.py

```diff
@@ -166,9 +169,26 @@ def serialized_landing_path(self):
             ]
 
     @property
-    def head_revision(self) -> str:
+    def landing_job_identifier(self) -> str:
         """Human-readable representation of the branch head's Phabricator revision ID."""
-        return f"D{self.revisions[-1].revision_id}"
+        if not self.revisions:
+            raise ValueError(
+                "Job must be associated with a revision to have a head revision."
+            )
+
+        head = self.revisions[-1]
+        if head.revision_id:
+            # Return the Phabricator identifier if the head revision has one.
+            return f"D{self.revisions[-1].revision_id}"
+
+        commit_message = head.patch_data.get("commit_message")
+        if commit_message:
+            # If there is no Phabricator identifier, return the first line of the
+            # commit message for the patch.
```

Comment:
Probably more useful to include the commit hash here? Or possibly both.

Reply:
The commit hash will change after landing, so the commit title would be a more useful identifier. This property is used as the identifying string in the landing failure email.

That said it just occurred to me that the head commit won't be a useful string either since most commit messages will just be something related to the try selector used, like [this push](https://hg.mozilla.org/try/rev/443c2cbbd878df752fc6833794a15e4126154272) for example. I'll update this to use the `-2` commit instead.